experimental/builder/include/ck_tile/builder/reflect/instance_traits_util.hpp

```diff
@@ -0,0 +1,196 @@
+    else if constexpr(std::is_same_v<T, ck::tensor_operation::element_wise::Relu>)
+        return "Relu";
+    else
+        return "unknown_op";
+}
+
+// Convert ConvolutionForwardSpecialization enum to string
+inline std::string
+conv_fwd_spec_name(ck::tensor_operation::device::ConvolutionForwardSpecialization spec)
+{
+    switch(spec)
+    {
+    case ck::tensor_operation::device::ConvolutionForwardSpecialization::Default: return "Default";
+    case ck::tensor_operation::device::ConvolutionForwardSpecialization::Filter1x1Stride1Pad0:
+        return "Filter1x1Stride1Pad0";
+    case ck::tensor_operation::device::ConvolutionForwardSpecialization::Filter1x1Pad0:
+        return "Filter1x1Pad0";
+    case ck::tensor_operation::device::ConvolutionForwardSpecialization::Filter3x3:
+        return "Filter3x3";
+    case ck::tensor_operation::device::ConvolutionForwardSpecialization::OddC: return "OddC";
+    }
+    return "unknown_conv_spec";
+}
+
+// Convert GemmSpecialization enum to string
+inline std::string gemm_spec_name(ck::tensor_operation::device::GemmSpecialization spec)
+{
+    switch(spec)
+    {
+    case ck::tensor_operation::device::GemmSpecialization::Default: return "Default";
+    case ck::tensor_operation::device::GemmSpecialization::MPadding: return "MPadding";
+    case ck::tensor_operation::device::GemmSpecialization::NPadding: return "NPadding";
+    case ck::tensor_operation::device::GemmSpecialization::KPadding: return "KPadding";
+    case ck::tensor_operation::device::GemmSpecialization::MNPadding: return "MNPadding";
+    case ck::tensor_operation::device::GemmSpecialization::MKPadding: return "MKPadding";
+    case ck::tensor_operation::device::GemmSpecialization::NKPadding: return "NKPadding";
+    case ck::tensor_operation::device::GemmSpecialization::MNKPadding: return "MNKPadding";
+    case ck::tensor_operation::device::GemmSpecialization::OPadding: return "OPadding";
+    case ck::tensor_operation::device::GemmSpecialization::MOPadding: return "MOPadding";
+    case ck::tensor_operation::device::GemmSpecialization::NOPadding: return "NOPadding";
+    case ck::tensor_operation::device::GemmSpecialization::KOPadding: return "KOPadding";
+    case ck::tensor_operation::device::GemmSpecialization::MNOPadding: return "MNOPadding";
+    case ck::tensor_operation::device::GemmSpecialization::MKOPadding: return "MKOPadding";
+    case ck::tensor_operation::device::GemmSpecialization::NKOPadding: return "NKOPadding";
+    case ck::tensor_operation::device::GemmSpecialization::MNKOPadding: return "MNKOPadding";
+    }
+    return "unknown_gemm_spec";
+}
+
+// Convert BlockGemmPipelineScheduler enum to string
+inline std::string pipeline_scheduler_name(ck::BlockGemmPipelineScheduler sched)
+{
+    switch(sched)
+    {
+    case ck::BlockGemmPipelineScheduler::Intrawave: return "Intrawave";
+    case ck::BlockGemmPipelineScheduler::Interwave: return "Interwave";
+    default: return "unknown_scheduler";
+    }
+}
```

Comment:
Since this is targeting C++20 anyway, this function can be slightly more readable (imo). This is mostly a nit though, feel free to ignore this suggestion, but I think it would be nice especially for the longer type names like `ck::tensor_operation::device::GemmSpecialization` etc.
```suggestion
{
    using enum ck::BlockGemmPipelineScheduler;
    switch(sched)
    {
    case Intrawave: return "Intrawave";
    case Interwave: return "Interwave";
    default: return "unknown_scheduler";
    }
}
```

Reply:
Also, you should probably leave the `default` case and put it below the switch (or consteval+throw, see previous), so that we can catch missing variants (-Wswitch is enabled in `cmake/EnableCompilerWarnings.cmake`). The `default:` case suppresses this warning.